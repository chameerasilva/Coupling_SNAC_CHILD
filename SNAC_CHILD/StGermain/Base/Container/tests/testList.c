/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003, Victorian Partnership for Advanced Computing (VPAC) Ltd, 110 Victoria Street, Melbourne, 3053, Australia.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Patrick D. Sunter, Software Engineer, VPAC. (pds@vpac.org)
**	Luke J. Hodkinson, Computational Engineer, VPAC. (lhodkins@vpac.org)
**	Siew-Ching Tan, Software Engineer, VPAC. (siew@vpac.org)
**	Alan H. Lo, Computational Engineer, VPAC. (alan@vpac.org)
**	Raquibul Hassan, Computational Engineer, VPAC. (raq@vpac.org)
**
**  This library is free software; you can redistribute it and/or
**  modify it under the terms of the GNU Lesser General Public
**  License as published by the Free Software Foundation; either
**  version 2.1 of the License, or (at your option) any later version.
**
**  This library is distributed in the hope that it will be useful,
**  but WITHOUT ANY WARRANTY; without even the implied warranty of
**  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
**  Lesser General Public License for more details.
**
**  You should have received a copy of the GNU Lesser General Public
**  License along with this library; if not, write to the Free Software
**  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
**
** $Id: testList.c 2136 2004-09-30 02:47:13Z PatrickSunter $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include <mpi.h>
#include "Base/Foundation/Foundation.h"
#include "Base/IO/IO.h"
#include "Base/Container/Container.h"

#include <stdio.h>
#include <stdlib.h>
#include <string.h>


int main( int argc, char* argv[] ) {
	MPI_Comm			CommWorld;
	int				rank;
	int				numProcessors;
	int				procToWatch;
	
	
	/* Initialise MPI, get world info */
	MPI_Init( &argc, &argv );
	MPI_Comm_dup( MPI_COMM_WORLD, &CommWorld );
	MPI_Comm_size( CommWorld, &numProcessors );
	MPI_Comm_rank( CommWorld, &rank );

	BaseFoundation_Init( &argc, &argv );
	BaseIO_Init( &argc, &argv );
	BaseContainer_Init( &argc, &argv );

	if( argc >= 2 ) {
		procToWatch = atoi( argv[1] );
	}
	else {
		procToWatch = 0;
	}
	
	if( rank == procToWatch ) {
		List*			list;
		Index			idx;
		Stream*			stream;
		
		stream = Journal_Register( Info_Type, "myStream" );
		
		list = List_New( sizeof(unsigned) );
		for( idx = 0; idx < 100; idx++ ) {
			List_Resize( list, List_Size( list ) + 1 );
			List_ElementAt( list, unsigned, idx ) = 100 - idx;
		}
		
		Print( list, stream );
		
		printf( "List Data:\n" );
		for( idx = 0; idx < 100; idx++ )
			printf( "\telements[%d]: %d\n", idx, List_ElementAt( list, unsigned, idx ) );
		
		Stg_Class_Delete( list );
	}
	
	BaseContainer_Finalise();
	BaseIO_Finalise();
	BaseFoundation_Finalise();
	
	/* Close off MPI */
	MPI_Finalize();
	
	return 0; /* success */
}


