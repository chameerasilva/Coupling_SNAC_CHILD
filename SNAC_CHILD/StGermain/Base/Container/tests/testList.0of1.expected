Stg_Class (ptr): 0x806a478
	sizeOfSelf: 52
	_deleteSelf: Yes
	type: List
	_delete (func ptr): 0x805787d
	_print (func ptr): 0x80578bd
	_copy (func ptr): (nil)
List (ptr): (0x806a478)
	elementSize: 4
	delta: 10
	maxElements: 100
	elementCnt: 100
	elements (ptr): 0x806a4b0
List Data:
	elements[0]: 100
	elements[1]: 99
	elements[2]: 98
	elements[3]: 97
	elements[4]: 96
	elements[5]: 95
	elements[6]: 94
	elements[7]: 93
	elements[8]: 92
	elements[9]: 91
	elements[10]: 90
	elements[11]: 89
	elements[12]: 88
	elements[13]: 87
	elements[14]: 86
	elements[15]: 85
	elements[16]: 84
	elements[17]: 83
	elements[18]: 82
	elements[19]: 81
	elements[20]: 80
	elements[21]: 79
	elements[22]: 78
	elements[23]: 77
	elements[24]: 76
	elements[25]: 75
	elements[26]: 74
	elements[27]: 73
	elements[28]: 72
	elements[29]: 71
	elements[30]: 70
	elements[31]: 69
	elements[32]: 68
	elements[33]: 67
	elements[34]: 66
	elements[35]: 65
	elements[36]: 64
	elements[37]: 63
	elements[38]: 62
	elements[39]: 61
	elements[40]: 60
	elements[41]: 59
	elements[42]: 58
	elements[43]: 57
	elements[44]: 56
	elements[45]: 55
	elements[46]: 54
	elements[47]: 53
	elements[48]: 52
	elements[49]: 51
	elements[50]: 50
	elements[51]: 49
	elements[52]: 48
	elements[53]: 47
	elements[54]: 46
	elements[55]: 45
	elements[56]: 44
	elements[57]: 43
	elements[58]: 42
	elements[59]: 41
	elements[60]: 40
	elements[61]: 39
	elements[62]: 38
	elements[63]: 37
	elements[64]: 36
	elements[65]: 35
	elements[66]: 34
	elements[67]: 33
	elements[68]: 32
	elements[69]: 31
	elements[70]: 30
	elements[71]: 29
	elements[72]: 28
	elements[73]: 27
	elements[74]: 26
	elements[75]: 25
	elements[76]: 24
	elements[77]: 23
	elements[78]: 22
	elements[79]: 21
	elements[80]: 20
	elements[81]: 19
	elements[82]: 18
	elements[83]: 17
	elements[84]: 16
	elements[85]: 15
	elements[86]: 14
	elements[87]: 13
	elements[88]: 12
	elements[89]: 11
	elements[90]: 10
	elements[91]: 9
	elements[92]: 8
	elements[93]: 7
	elements[94]: 6
	elements[95]: 5
	elements[96]: 4
	elements[97]: 3
	elements[98]: 2
	elements[99]: 1
